apps/designer/app/designer/features/style-panel/controls/grid-control.tsx

```diff
@@ -0,0 +1,154 @@
+  const setJustifyContent = batchUpdate.setProperty("justifyContent");
+  return (
+    <Grid
+      data-property={name}
+      css={{
+        width: "100%",
+        aspectRatio: "1 / 1",
+        padding: "6px",
+        borderRadius: "4px",
+        background: "$loContrast",
+        border: "2px solid currentColor",
+        alignItems: "center",
+        gap: "1px",
+        gridArea: name,
+        gridTemplateColumns: "repeat(3, 1fr)",
+        gridTemplateRows: "repeat(3, 1fr)",
+        gridTemplateAreas: `
+          "${cells.slice(0, 0 + 3).join(" ")}"
+          "${cells.slice(3, 3 + 3).join(" ")}"
+          "${cells.slice(6, 6 + 3).join(" ")}"
+        `,
+        color: isCurrentBreakpoint ? "$colors$blue9" : "$colors$slate8",
+        transform: direction
+          ? "rotate(-90deg) scaleX(-1)"
+          : "rotate(0deg) scaleX(1)",
+      }}
+    >
+      {cells.map((value, index) => (
+        <Flex
+          justify="center"
+          align="center"
+          key={index}
+          css={{
+            gridArea: value,
+            width: "100%",
+            height: "100%",
+          }}
+        >
+          <IconButton
+            css={{
+              width: "100%",
+              height: "100%",
+              color: "$colors$gray8",
+              "&:focus": { boxShadow: "none" },
+            }}
+            onClick={() => {
+              const [alignItems, justifyContent] = cells[index]
+                .slice(1)
+                .split("_")
+                .map((value) => ["start", "center", "end"][parseFloat(value)]);
+              setAlignItems(alignItems);
+              setJustifyContent(justifyContent);
+              batchUpdate.publish();
+            }}
+          >
+            <DotFilledIcon />
+          </IconButton>
+        </Flex>
+      ))}
+      <Flex
+        css={{
+          gridArea: cells[currentPosition],
+          ...(alignItems === "stretch" && {
+            gridColumnEnd: "none",
+          }),
+          ...(justifyContent === "space-around" && {
+            gridRowEnd: "none",
+            justifyContent: "space-around",
+          }),
+          alignItems:
+            `start center end start center end start center end`.split(" ")[
+              currentPosition
+            ],
+          pointerEvents: "none",
+          flexDirection: "column",
+          width: "100%",
+          height: "100%",
+          gap: "calc($sizes$1 / 2)",
+          color: "currentColor",
+        }}
+      >
+        {["60%", "100%", "60%"].map((value, index) => (
+          <Flex
+            key={index}
+            css={{
+              inlineSize: value,
+              blockSize: "calc(100% / 3)",
+              maxBlockSize: "6px",
+              background: "currentColor",
+              borderRadius: "calc($radii$1 / 2)",
+            }}
+          ></Flex>
+        ))}
+      </Flex>
+    </Grid>
+  );
+};
+
+export { GridControl };
```

Comment:
Nit: would rather export right where the component is defined to see right away that its exported without scrolling